New code generator fails (2.1-RC2)

Using the new generator, one of my project fails with the following error:

>  Using "Exec" task from assembly "Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".
>        Task "Exec"
>          "/usr/share/dotnet/dotnet" "/home/sro/.nuget/packages/microsoft.orleans.codegenerator.msbuild/2.1.0-rc2/build/../tasks/netcoreapp2.0/Orleans.CodeGenerator.MSBuild.dll" SourceToSource "/home/sro/projects/xxx/src/Core.Grains/obj/Debug/netstandard2.0/Core.Grains.orleans.g.args.txt"
>          Orleans.CodeGenerator - command-line = SourceToSource /home/sro/projects/xxx/src/Core.Grains/obj/Debug/netstandard2.0/Core.Grains.orleans.g.args.txt
>          -- Code Generation FAILED -- 
>          
>          Exc level 0: System.InvalidOperationException: Unable to find type with metadata name "System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverageAttribute".
>             at Orleans.CodeGenerator.WellKnownTypes.<FromCompilation>g__Type|0_0(String type, <>c__DisplayClass0_0& )
>             at Orleans.CodeGenerator.WellKnownTypes.FromCompilation(Compilation compilation)
>             at Orleans.CodeGenerator.CodeGenerator..ctor(Compilation compilation, ILogger log)
>             at Orleans.CodeGenerator.MSBuild.CodeGeneratorCommand.<Execute>d__39.MoveNext()
>          --- End of stack trace from previous location where exception was thrown ---
>             at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>             at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>             at Microsoft.Orleans.CodeGenerator.MSBuild.Program.SourceToSource(String[] args)
>             at Microsoft.Orleans.CodeGenerator.MSBuild.Program.Main(String[] args)
>    1:7>/home/sro/.nuget/packages/microsoft.orleans.codegenerator.msbuild/2.1.0-rc2/build/Microsoft.Orleans.CodeGenerator.MSBuild.targets(122,5): error MSB3073: The command ""/usr/share/dotnet/dotnet" "/home/sro/.nuget/packages/microsoft.orleans.codegenerator.msbuild/2.1.0-rc2/build/../tasks/netcoreapp2.0/Orleans.CodeGenerator.MSBuild.dll" SourceToSource "/home/sro/projects/xxx/src/Core.Grains/obj/Debug/netstandard2.0/Core.Grains.orleans.g.args.txt"" exited with code 3. [/home/sro/projects/xxx/src/Core.Grains/Core.Grains.csproj]
>        Done executing task "Exec" -- FAILED.

In total, I have 4 projects using the generator and only 1 fails.
I use dotnet 2.1.402

I don't know how to provide you more information about the problem.

--- a/src/Orleans.CodeGenerator.MSBuild/CodeGeneratorCommand.cs
+++ b/src/Orleans.CodeGenerator.MSBuild/CodeGeneratorCommand.cs
@@ -84,8 +84,17 @@ namespace Orleans.CodeGenerator.MSBuild
                 ? ProjectId.CreateFromSerialized(projectIdGuid)
                 : ProjectId.CreateNewId();
 
+            this.Log.LogDebug($"AssemblyName: {this.AssemblyName}");
             this.Log.LogDebug($"ProjectGuid: {ProjectGuid}");
             this.Log.LogDebug($"ProjectID: {projectId}");
+            this.Log.LogDebug($"ProjectName: {projectName}");
+            this.Log.LogDebug($"CodeGenOutputFile: {this.CodeGenOutputFile}");
+            this.Log.LogDebug($"ProjectPath: {this.ProjectPath}");
+            this.Log.LogDebug($"OutputType: {this.OutputType}");
+            this.Log.LogDebug($"TargetPath: {this.TargetPath}");
+            this.Log.LogDebug($"DefineConstants ({this.DefineConstants.Count}): {string.Join(", ", this.DefineConstants)}");
+            this.Log.LogDebug($"Sources ({this.Compile.Count}): {string.Join(", ", this.Compile)}");
+            this.Log.LogDebug($"References ({this.Reference.Count}): {string.Join(", ", this.Reference)}");
 
             var languageName = GetLanguageName(ProjectPath);
 

--- a/src/Orleans.CodeGenerator/WellKnownTypes.cs
+++ b/src/Orleans.CodeGenerator/WellKnownTypes.cs
@@ -103,16 +103,36 @@ namespace Orleans.CodeGenerator
 
             INamedTypeSymbol Type(string type)
             {
-                var result = compilation.GetTypeByMetadataName(type);
-                if (result == null) throw new InvalidOperationException($"Unable to find type with metadata name \"{type}\".");
+                var result = ResolveType(type);
+                if (result == null)
+                {
+                    throw new InvalidOperationException($"Unable to find type with metadata name \"{type}\".");
+                }
                 return result;
             }
 
             OptionalType OptionalType(string type)
             {
-                var result = compilation.GetTypeByMetadataName(type);
+                var result = ResolveType(type);
                 if (result == null) return None.Instance;
                 return new Some(result);
+            }
+
+            INamedTypeSymbol ResolveType(string type)
+            {
+                var result = compilation.GetTypeByMetadataName(type);
+                if (result == null)
+                {
+                    foreach (var reference in compilation.References)
+                    {
+                        var asm = compilation.GetAssemblyOrModuleSymbol(reference) as IAssemblySymbol;
+                        if (asm == null) continue;
+                        result = asm.GetTypeByMetadataName(type);
+                        if (result != null) break;
+                    }
+                }
+
+                return result;
             }
         }
 

